packages/indexer-generator/src/render-domain.ts

```diff
@@ -306,21 +326,16 @@ export default class MainDomain
     return { info, stats }
   }
 
-  async getAccountEventsByTime(
-    account: string,
-    startDate: number,
-    endDate: number,
-    opts: any,
-  ): Promise<StorageStream<string, ${Name}Event>> {
-    const stream = await this.context.apiClient
+  async getAccountEvents(args: EventsFilters): Promise<${Name}Event[]> {
+    const response = await this.context.apiClient
       .useBlockchain(BlockchainChain.Solana)
       .invokeDomainMethod({
-        account,
-        method: 'getAccountEventsByTime',
-        args: [startDate, endDate, opts],
+        account: args.account,
+        method: 'getAccountEvents',
+        args: [args],
       })
 
-    return stream as StorageStream<string, ${Name}Event>
```

Comment:
Why aren't we using `StorageStream` anymore?

Reply:
moved the logic here: https://github.com/aleph-im/aleph-indexer-library/blob/a7c115ffe3527464fdd207021da6c7da8dc133b3/packages/marinade-finance/src/domain/account.ts#L37, where the stream is consumed, now is also processed there and returns directly the parsed type to not propagate the stream to caller functions

Reply:
Hm, makes sense to stop it at the domain. Good catch :+1: